packages/firebase_dynamic_links/lib/src/dynamic_link_parameters.dart

```diff
@@ -0,0 +1,110 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of firebase_dynamic_links;
+
+/// The class used for Dynamic Link URL generation.
+///
+/// Supports creation of short and long Dynamic Link URLs. Short URLs will have
+/// a domain and a randomized path. Long URLs will have a domain and a query
+/// that contains all of the Dynamic Link parameters.
+class DynamicLinkParameters {
+  DynamicLinkParameters({
+    this.androidParameters,
+    @required this.domain,
+    this.dynamicLinkParametersOptions,
+    this.googleAnalyticsParameters,
+    this.iosParameters,
+    this.itunesConnectAnalyticsParameters,
+    @required this.link,
+    this.navigationInfoParameters,
+    this.socialMetaTagParameters,
+  });
```

Comment:
You should really add `}) : assert(domain != null), assert(link != null);` here. The `@required` annotation yields an analyzer error, if you leave out the parameter at a call site, not if you explicitly set it to `null`.

Same in other classes.